internal/plugin.go

```diff
@@ -329,6 +329,17 @@ func (p *Plugin) applyDefaults(unmarshaledConfig map[string]interface{}) {
 		p.PolicyDefaults.Controls = defaults.Controls
 	}
 
+	if len(p.PolicyDefaults.PolicyAnnotations) > 0 {
+		annotation := p.PolicyDefaults.PolicyAnnotations
+		for i := range p.Policies {
```

Comment:
Could you please reuse the `for i := range p.Policies {` loop below?

Also, like I said, could you only apply `p.PolicyDefaults.PolicyAnnotations` to the policy if the policy doesn't define a map. It shouldn't be a union. Also, if the policy has explicitly set `PolicyAnnotations` to an empty map, it shouldn't inherit the defaults.

Reply:
As a user when I define  `p.PolicyDefaults.PolicyAnnotations` that is mean I need to have this annotations in all generated policies from this policygenerator template. 
If there are some annotations I don't want them in all generated policies I would not add them in `p.PolicyDefaults.PolicyAnnotations` I add them to the desired `PolicyConfig.PolicyAnnotations` is that make sense ? do I miss understand ?

Reply:
I see your point, but after discussing with the team, I don't want the override/inheritance behavior to be different for this field compared to the other policy defaults.